exporters/prometheus/exporter_test.go

```diff
@@ -386,3 +386,178 @@ func TestMultiScopes(t *testing.T) {
+
+				barB, err := meterB.SyncInt64().Counter("bar",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter b bar"))
+				assert.NoError(t, err)
+				barB.Add(ctx, 100, attribute.String("type", "bar"))
+			},
+			possibleExpectedFiles: []string{
+				"testdata/conflict_help_counter_1.txt",
+				"testdata/conflict_help_counter_2.txt",
+			},
+		},
+		{
+			name: "conflict_help_updowncounter",
+			recordMetrics: func(ctx context.Context, meterA, meterB otelmetric.Meter) {
+				barA, err := meterA.SyncInt64().UpDownCounter("bar",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter a bar"))
+				assert.NoError(t, err)
+				barA.Add(ctx, 100, attribute.String("type", "bar"))
+
+				barB, err := meterB.SyncInt64().UpDownCounter("bar",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter b bar"))
+				assert.NoError(t, err)
+				barB.Add(ctx, 100, attribute.String("type", "bar"))
+			},
+			possibleExpectedFiles: []string{
+				"testdata/conflict_help_updowncounter_1.txt",
+				"testdata/conflict_help_updowncounter_2.txt",
+			},
+		},
+		{
+			name: "conflict_help_histogram",
+			recordMetrics: func(ctx context.Context, meterA, meterB otelmetric.Meter) {
+				barA, err := meterA.SyncInt64().Histogram("bar",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter a bar"))
+				assert.NoError(t, err)
+				barA.Record(ctx, 100, attribute.String("A", "B"))
+
+				barB, err := meterB.SyncInt64().Histogram("bar",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter b bar"))
+				assert.NoError(t, err)
+				barB.Record(ctx, 100, attribute.String("A", "B"))
+			},
+			possibleExpectedFiles: []string{
+				"testdata/conflict_help_histogram_1.txt",
+				"testdata/conflict_help_histogram_2.txt",
+			},
+		},
+		{
+			name: "conflict_unit",
+			recordMetrics: func(ctx context.Context, meterA, meterB otelmetric.Meter) {
+				bazA, err := meterA.SyncInt64().Counter("bar",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter bar"))
+				assert.NoError(t, err)
+				bazA.Add(ctx, 100, attribute.String("type", "bar"))
+
+				bazB, err := meterB.SyncInt64().Counter("bar",
+					instrument.WithUnit(unit.Milliseconds),
+					instrument.WithDescription("meter bar"))
+				assert.NoError(t, err)
+				bazB.Add(ctx, 100, attribute.String("type", "bar"))
+			},
+			options:               []Option{WithoutUnits()},
+			possibleExpectedFiles: []string{"testdata/conflict_unit.txt"},
+		},
+		{
+			name: "conflict_type_counter",
+			recordMetrics: func(ctx context.Context, meterA, meterB otelmetric.Meter) {
+				counter, err := meterA.SyncInt64().Counter("foo",
+					instrument.WithUnit(unit.Bytes),
+					instrument.WithDescription("meter foo"))
+				assert.NoError(t, err)
+				counter.Add(ctx, 100, attribute.String("type", "foo"))
+
+				gauge, err := meterA.SyncInt64().UpDownCounter("foo_total",
```

Comment:
Why are the description outputs non-deterministic and there needs to be two `possibleExpectedFiles`, but this is deterministic?

Reply:
Good point, I will add another possible case.